src/primitives/transaction.cpp

```diff
@@ -223,3 +221,32 @@ std::string CTransaction::ToString() const
         str += "    " + vout[i].ToString() + "\n";
     return str;
 }
+
+// Update the txid which is non-malleable.
+// Signature data is cleared before the transaction is serialized and hashed.
+void CTransaction::UpdateTxid() const
+{
+    // Create a deep copy of this transaction
+    CMutableTransaction tx(*this);
+
+    // We keep the sigscript for coinbase txs to avoid duplicate txids (BIP34 and BIP30)
+    if (!IsCoinBase()) {
+        // Clear sigscript from all transaction inputs.
+        for (CTxIn & txIn : tx.vin) {
+            txIn.scriptSig.clear();
+        }
+
+        // Clear joinSplitSig by filling the buffer with zero
+        tx.joinSplitSig.assign(0);
+    }
+
+    *const_cast<uint256*>(&txid) = SerializeHash(tx);
```

Comment:
Does this mean that we aren't actually "skipping" the sigs as such, but instead are including them as zeroes? That needs to be clear for people to recreate the txids.